umpost/um2netcdf.py

```diff
@@ -307,36 +307,16 @@ def process(infile, outfile, args):
 
     cubes = iris.load(infile)
     cubes.sort(key=lambda cs: cs.attributes['STASH'])
+    cube_lookup = {to_item_code(c.attributes['STASH']): c for c in cubes}
```

Comment:
One other way we could do this is to to add item codes directly to the cubes as  attributes, e.g:
```python
for c in cubes:
   c.attributes['item_code'] = to_item_code(c.attributes['STASH'])
```
which can then be accessed later on by `check_pressure_level_masking`

Reply:
That looks good to me!

Reply:
The initial dict version is a touch simpler as the key/item code & cube are nicely paired. Moving item code into the `Stash` keeps things together, but it's a tad odd putting attributes into a sub data structure, requiring a custom `DummyCube` for testing.

I'll examine the rest of the script today to inform if one way is neater than another...

Reply:
> One other way we could do this is to to add item codes directly to the cubes as attributes, e.g:
> ```python
> for c in cubes:
>    c.attributes['item_code'] = to_item_code(c.attributes['STASH'])
> ```
> which can then be accessed later on by `check_pressure_level_masking`

This changes cube _global_ attributes, which are written to _global_ attributes in NetCDFs. This adds a field for the test case, which "breaks" binary file equivalence between the repo's first version & [this version](https://github.com/ACCESS-NRI/um2nc-standalone/commit/055568f9abfc1ef12458d8edd46a079e7121174f).

l'll attempt something else, with a backup being to revert to the `{item_code: cube}` dict style.

Reply:
Ah I hadn't thought about that. So any changes to the attributes end up being written to the NetCDFs? I see why that might not be the ideal place for temporary processing information now.

Reply:
Yep, all attrs end up in the NC file. As a test, I added another random str key & ran against the test file:
```
 // global attributes:
    :Conventions = "CF-1.6" ;
    :item_code = 5216ll ;
    :item_code_FAKE = "HERE IS FAKE" ;
    :source = "Data from Met Office Unified Model" ;
    :um_version = "7.3" ;
```
The junk attr made it in...

Reply:
@blimlim how do you find the readability of [adding item_code directly to the object](https://github.com/ACCESS-NRI/um2nc-standalone/commit/9a076ece8ac252ae060dfe139072662c9e678b6f)?

It might look better in a side by side diff view...

Reply:
I think that looks ok, and trust your judgment on whether this or the `cube_lookup` is a better here

Reply:
With `item_code` out an attr `dict`, the `DummyCube` for testing got simpler too.

Given the code will evolve over time, changes may be needed as more of the utility is picked apart. It's easy enough to change though.